data/com.endlessm.DemoMode.rules

```diff
@@ -0,0 +1,10 @@
+polkit.addRule(function(action, subject) {
+    if ((action.id == "org.freedesktop.Flatpak.app-install" ||
+         action.id == "org.freedesktop.Flatpak.runtime-install"||
+         action.id == "org.freedesktop.Flatpak.app-uninstall" ||
+         action.id == "org.freedesktop.Flatpak.runtime-uninstall") &&
+        subject.active == true && subject.local == true &&
+        subject.user == "demo-guest") {
```

Comment:
What about the case where someone creates a `demo-guest` user by hand but the system is not in demo mode? I see that the GDM hooks [check a stamp file](https://github.com/endlessm/gdm/blob/master/data/PostSession.in#L12-L14):

```
DEMO_MODE_STAMP=/run/gnome-initial-setup/eos-demo-mode

if [ -f ${DEMO_MODE_STAMP} ] && [ $(grep -E "^${USERNAME}$" ${DEMO_MODE_STAMP}) ] ; then
```

You could use [polkit.spawn](https://www.freedesktop.org/software/polkit/docs/master/polkit.8.html) in the of the condition:

```js
try {
    polkit.spawn(["/bin/test", "-e", "/run/gnome-initial-setup/eos-demo-mode"]);
    return polkit.Result.YES;
} catch (error) { /* no-op */ }
```

which would avoid the overhead of spawning a process unless the username really is `demo-guest`. Or… can we add this policy at run-time when demo mode is enabled?

It would be nice not to need to run `polkit.spawn()` for every Flatpak action though…

Reply:
I created a standard user "demo-guest" on my master system (with that polkit rule installed). After logging in as demo-guest I can not install applications, still need authentication as an admin.

Reply:
Okay, so (why) does it work in demo mode?

Reply:
Ok, actually I double checked and gnome-add-user was not doing what I was expecting. I created a user now with adduser and indeed I can install apps with that user. So, we should handle the case from above.

Reply:
OK, added a check for this.